src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const x = this[i];
+
+            this[i] = this[j];
+            this[j] = x;
+          }
+        } else {
+          if (String(this[j]) < String(this[i])) {
+            const x = this[i];
+
+            this[i] = this[j];
+            this[j] = x;
+          }
+        }
```

Comment:
The 'compareFunction' should return 0 for equal elements. It's important to handle this case explicitly to ensure stability of the sort (i.e., equal elements retain their original order).